File: app/src/main/java/com/mercandalli/tracker/common/Closer.java
package com.mercandalli.tracker.common;

import java.io.Closeable;

public class Closer {

    public static void closeSilently(Closeable... xs) {
        for (Closeable x : xs) {
            if (x != null) {
                try {
                    x.close();
                } catch (Throwable ignored) {
                }
            }
        }
    }
}

File: app/src/main/java/com/mercandalli/tracker/common/Closer.kt
package com.mercandalli.tracker.common

import java.io.Closeable

internal object Closer {

    fun closeSilently(vararg xs: Closeable?) {
        xs.filterNotNull().forEach {
            try {
                it.close()
            } catch (ignored: Throwable) {
            }
        }
    }
}
